vscode/microsoft-kiota/src/kiotaInterop.ts

```diff
@@ -7,8 +7,9 @@ import { getWorkspaceJsonDirectory } from "./util";
 export async function connectToKiota<T>(context: vscode.ExtensionContext, callback:(connection: rpc.MessageConnection) => Promise<T | undefined>): Promise<T | undefined> {
   const kiotaPath = getKiotaPath(context);
   await ensureKiotaIsPresent(context);
+  const workspaceDirectory = getWorkspaceJsonDirectory();
```

Comment:
sorry, I wasn't specific enough in my comment. we should wrap that call with a "get parent directory call" since the cwd you want is the parent directory of ".kiota"

Reply:
I've updated the workspaceDirectory function instead

Reply:
I think the update you've made is progress. And we need to go further by splitting the function in 2: getWorkspaceJsonDirectory, getWorkspaceRootDirectory.
The first one calling into the second function and adding the `.kiota` segment.
This way here you can call the second one.
Does that make sense?

Reply:
Yeah this makes sense! 
I'll update this

Reply:
So I was creating a `/kiota` directory instead and setting that as the default base directory, so that if no workspace folder exists then the `kiota` directory will be created that will contain everything (i.e the `.kiota` folder and the output files)
What do you think?

Reply:
Oh, you want to create a subdirectory for the current working directory when no workspace is open? This way we would not be emitting everything into the home directory directly but in a subdirectory? But then the rest of the structure would remain the same. As far as I understand your comment, correct? 
If we have the same understanding, yes, I think this makes sense. This will avoid making a mess in the home directory of the user. 
If this is not the same thing we're talking about, let's talk about this further

Reply:
Just to confirm, this would end up in ~/kiota/project-01 for instance?

Reply:
Okay based on our earlier conversation...
I can use `~/kiota` as the base directory where the workspace files go and then the output can go to `~/kiota/{clientOrPluginName}`
What are your thoughts on this?

Reply:
so we'd have (when no workspace is open):
- cwd: `~/kiota`
- worskspace dir: `~/kiota/.kiota`
- client/plugin dir: `~/kiota/clientName` (because we default the output path in that scenario`

(repeating to make sure we're all on the same page here)

Reply:
yeah! This works